### ISSUE TITLE
Image description: Pre-populate from IPTC data

### ISSUE THREAD

**Issue by fwhamm** [mastodon/mastodon#20297]
### Pitch

Mastodon faves adding image descriptions. Often the image description is already available as IPTC data in the field "Headline". Many people, especially photographers, edit IPTC data for their photos, so those have to redundantly enter image description again when uploading an imapge to Mastodon.

I suggest pre-populating the image description from the IPTC field "Headline" when uploading (like Wordpress does this since several years). Or at least offering a button to populate it on demand.

### Motivation

Since I am a photographer, almost all photos I upload from my notebook have the IPTC data already edited, especially "Headline". So I have to redundantly enter the image description every time, even though it is already included in the IPTC data.

I discussed this issue with other photographers and they would appreciate this feature. Fore example there is a bunch of photogaphers connecting (previously on Twitter) with hashtag #Fotomontag ("photo monday"). On mondays they each feature one of their photo. Those photos are mostly have theire IPTC data well edited, but image description has manually to be entered.

**Comment by ineffyble** [comment]
This data gets stripped from uploaded media, so there would be need to be a process to extract it first and make it available to the frontend somehow 🤔

**Comment by ClearlyClaire** [comment]
Note that the server itself often gets a version pre-processed by the client: the Web client handles resizing if needed and possible. So I guess it would make sense to do it in the client if possible.

**Comment by erbridge** [comment]
https://github.com/MikeKovarik/exifr looks like a handy library for doing the parsing: `exifr.parse(file, { iptc: true })`

**Comment by fwhamm** [comment]
Sure, this could be implemented from each separate client. What about the web app? As far as I can see pictures are uploaded first to the server. Then it is possible to edit the description. And there is a link to extract text from the picture via OCR. Same could hopefully be done for extracting EXIF.

(I normally use Mastadon via the web app in the browser)

**Comment by ineffyble** [comment]
Looking at the code, the web client already reads EXIF data from the original image to make sure it is rotated correctly. So it shouldn't be hard to extract the IPTC fields and pre-populate description.

**Comment by talios** [comment]
> This data gets stripped from uploaded media, so there would be need to be a process to extract it first and make it available to the frontend somehow 🤔

@ineffyble Does that mean Mastodon strips off any copyright information in posted images? For photography, that's less than ideal.

**Comment by ineffyble** [comment]
Yes, most social networks and chat apps strip image metadata for privacy
reasons, since many photos include a lot of metadata, often including GPS
coordinates of where they were taken.

**Comment by akwala** [comment]
> I suggest pre-populating the image description from the IPTC field "Headline" when uploading (like Wordpress does this since several years). Or at least offering a button to populate it on demand.

[IPTC:Description](https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#description) is more in line with the aim of accessibility. This the Help Text for IPTC:Description from the[ IPTC specification](https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata):

> Enter a "caption" describing the who, what, and why of what is happening in this image, this might include names of people, and/or their role in the action that is taking place within the image.

Going by the [usage examples](https://www.iptc.org/std/photometadata/documentation/userguide/#_metadata_usage_examples) of the IPTC spec, IPTC:Headline is a better match for the `title` attribute.

IPTC:Description [maps to Exif:Description](https://iptc.org/std/photometadata/documentation/mappingguidelines/#description) and the [IPTC mapping guidelines note](https://iptc.org/std/photometadata/documentation/mappingguidelines/#exif-note-on-description) the following:

> The widely used Exif Tag for the description of what an image shows is the Exif ImageDescription.

[IPTC:Headline](https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#headline), on the other hand, [does not map to any Exif field](https://iptc.org/std/photometadata/documentation/mappingguidelines/#headline). 

Is there a standard mapping of IPTC/Exif fields/tags to the [`alt`](https://www.w3.org/TR/2011/WD-html5-author-20110809/the-img-element.html#attr-img-alt) and [`title`](https://www.w3.org/TR/2011/WD-html5-author-20110809/global-attributes.html#the-title-attribute) attributes of `<img />`?

**Comment by nuclearfall** [comment]
I agree this would help with accessibility. It's especially helpful when an image is reused frequently. I could add alt text to images in advance.

**Comment by zeroexp** [comment]
Hello - I am a member of the IPTC photo-metadata working group and I also created a system that emphasizes a workflow from embedded metadata all the way to structured data and publishing with the same data (ImageSnippets). You can add the alt-text in ImageSnippets or import it from other sources and then you can share the image in an html file (see below), but with all metadata converted into linked data and schema.org.  We also store the alt-text data as the alt attribute for the img tag in the html. 

When we read in images - if no alt-text exists, we read in the Headline field as sometime description can be quite long and include information that is less useful and quite different from alt-text. We did a lot of experimentation before deciding on this process . 

Here is the IPTC reference standard image. You can see all of the IPTC metadata at this link: https://imagesnippets.com/imgtag/images/info@margaretwarren.us/IPTC-PhotometadataRef-Std2022.1.html    ---- but also, if you take any image and go to https://getpmd.iptc.org/getiptcpmd.html -- you can see what iptc data is embedded in any images. Of course you can also use various ExifViewers to also look at data in the image.  

Many people have privacy concerns about location data for very legitimate and scary reasons... but I like the way iNaturalist handles it. If your image contains location data they ask if you would like it opaque in their system (i.e. not deleted, but not able to see in their UI), or deleted or retained. 

I think a lot of embedded metadata can be retained for re-use while giving users a decision about their privacy concerns. For me personally, I don't usually share images liberally in social media of events that are in the present moment. I make art and thus tend to share things after they have been processed and I want to leave location data - but I can understand why people want it removed. 

I've always thought it would not hurt for people to just have a tiny bit of friction in the publishing process and be given the choice to disable location data rather than just stripping everything out. There are many people who don't want to consider any embedded data, but there are many others who want to deliberately add it during a larger workflow.

**Comment by akwala** [comment]
Related:  Pre-populate the image alt text field from the description field in EXIF metadata #14903

**Comment by theotherlinh** [comment]
Definitely would like to see data pulled from IPTC to prefill a post and especially for the newer altTextAccessibility field. Adobe Bridge supports that field so I'm hoping to see it come to LR in the near future. But it would be nice to write-once copy-anywhere.

**Comment by alexisart** [comment]
+1 Would also like to see this implemented. If you are looking for the description of the related fields, see https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#alt-text-accessibility and https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#extended-description-accessibility

**Comment by D4llo** [comment]
[exifreader](https://www.npmjs.com/package/exifreader) can also be a solution to red IPTC.

Another thing handy, I think, would be to write the post's image description into the image's AltTextAccessibility metada on download if the field is empty.

So if we share the image later on, we can still reuse the alt.

Note that:

> This property should not be confused with the IPTC property Headline, which is a brief synopsis or summary of the contents of the image. This property is required for conformance with the W3C Web Content Accessibility Guidelines (WCAG).


https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#alt-text-accessibility
